### PR TITLE
fix (CropperActivity) sort corners after the user modified their position

### PR DESCRIPTION
**Pull request by giovanni-nic-kuama**
fix (BaseScannerActivity) manage device orientation togheter to Bitmap orientation

### CODE REVIEW

File: app/src/main/java/net/kuama/documentscanner/viewmodels/CropperModel.kt [hunk 1]
```diff
@@ -42,15 +44,19 @@ class CropperModel : ViewModel() {
...
+            val acceptedAndOrderedCorners = listOf(acceptedCorners.topLeft, acceptedCorners.topRight,
+                acceptedCorners.bottomRight, acceptedCorners.bottomLeft)
+            viewModelScope.launch {
+                perspectiveTransform(
+                    PerspectiveTransform.Params(
+                        bitmap = bitmap,
+                        corners = CornersFactory.create(acceptedAndOrderedCorners, acceptedCorners.size)
+                    )
+                ) { result ->
+                    result.fold(::handleFailure) { bitmap ->
```
Comment (GabM3) [inline app/src/main/java/net/kuama/documentscanner/viewmodels/CropperModel.kt]:
```suggestion
                ) {  
                    it.fold(::handleFailure) { bitmap ->
```

**Review by GabM3 — CHANGES_REQUESTED** [review]

**Review by GabM3 — DISMISSED** [review]

File: app/src/main/java/net/kuama/documentscanner/presentation/CropperActivity.kt [hunk 2]
```diff
@@ -24,6 +24,7 @@ import java.util.*
 class CropperActivity : AppCompatActivity() {
     private lateinit var cropModel: CropperModel
     private lateinit var bitmapUri: Uri
+    private var screenOrientationDeg: Int? = 0
```
Comment (matteoKuama) [inline app/src/main/java/net/kuama/documentscanner/presentation/CropperActivity.kt]:
```suggestion
    private var screenOrientationDeg: Int = 0
```
or
```suggestion
    private var screenOrientationDeg: Int?
```
choose nullable or with default value

File: app/src/main/java/net/kuama/documentscanner/viewmodels/CropperModel.kt [hunk 3]
```diff
@@ -22,12 +23,13 @@ class CropperModel : ViewModel() {
     val originalBitmap = MutableLiveData<Bitmap>()
     val bitmapToCrop = MutableLiveData<Bitmap>()
 
-    fun onViewCreated(uri: Uri, contentResolver: ContentResolver) {
+    fun onViewCreated(uri: Uri, screenOrientationDeg: Int?, contentResolver: ContentResolver) {
```
Comment (matteoKuama) [inline app/src/main/java/net/kuama/documentscanner/viewmodels/CropperModel.kt]: if you choose to have a default value
```suggestion
    fun onViewCreated(uri: Uri, screenOrientationDeg: Int, contentResolver: ContentResolver) {
```

File: app/src/main/java/net/kuama/documentscanner/viewmodels/CropperModel.kt [hunk 4]
```diff
@@ -42,15 +44,19 @@ class CropperModel : ViewModel() {
...
-                PerspectiveTransform.Params(
-                    bitmap = bitmap,
-                    corners = corners.value!!
-                )
-            ) { result ->
-                result.fold(::handleFailure) { bitmap ->
-                    bitmapToCrop.value = bitmap
+        corners.value?.let { acceptedCorners ->
+            val acceptedAndOrderedCorners = listOf(acceptedCorners.topLeft, acceptedCorners.topRight,
+                acceptedCorners.bottomRight, acceptedCorners.bottomLeft)
```
Comment (matteoKuama) [inline app/src/main/java/net/kuama/documentscanner/viewmodels/CropperModel.kt]:
```suggestion
            val acceptedAndOrderedCorners = listOf(acceptedCorners.topLeft,
                                                                              acceptedCorners.topRight,
                                                                              acceptedCorners.bottomRight, 
                                                                              acceptedCorners.bottomLeft,
                                                                             )
```
format ?

File: app/src/main/java/net/kuama/documentscanner/presentation/BaseScannerActivity.kt [hunk 5]
```diff
@@ -98,19 +101,41 @@ abstract class BaseScannerActivity : AppCompatActivity() {
...
 
+    private val orientationEventListener by lazy {
+        object : OrientationEventListener(this) {
+            override fun onOrientationChanged(orientation: Int) {
+                if (orientation == ORIENTATION_UNKNOWN) {
+                    return
+                }
+
+                val rotationDegree = when (orientation) {
+                    in 45 until 135 -> 270
```
Comment (matteoKuama) [inline app/src/main/java/net/kuama/documentscanner/presentation/BaseScannerActivity.kt]:
```suggestion
               val rotationDegree = when (orientation) {
                    ORIENTATION_UNKNOWN -> return
                    in 45 until 135 -> 270
```

**Review by matteoKuama — CHANGES_REQUESTED** [review]

**Review by matteoKuama — COMMENTED** [review]
rebase

**Review by matteoKuama — APPROVED** [review]

**Pull request merged**